atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/SnapshotStoreImpl.java

```diff
@@ -27,23 +29,36 @@
 import java.util.Collection;
 import java.util.Comparator;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.NavigableMap;
 import java.util.Optional;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
 import javax.annotation.concurrent.NotThreadSafe;
 
 @NotThreadSafe
 final class SnapshotStoreImpl implements SnapshotStore {
     private static final SafeLogger log = SafeLoggerFactory.get(SnapshotStoreImpl.class);
-    private static final int MAXIMUM_SIZE = 20_000;
+    private static final Sequence MAX_VERSION = Sequence.of(Long.MAX_VALUE);
 
-    private final Map<Sequence, ValueCacheSnapshot> snapshotMap;
-    private final SetMultimap<Sequence, StartTimestamp> liveSequences;
```

Comment:
I don't know if it mattered that this was a set multimap or not, really (or rather, enough that I needed to keep it typed)